src/python/pants/backend/python/rules/run_setup_py.py

```diff
@@ -272,40 +287,42 @@ async def run_setup_pys(
     # globbed over, but erroring on any explicitly-requested non-exported targets.
 
     exported_targets: List[ExportedTarget] = []
-    explicit_nonexported_targets: List[HydratedTarget] = []
-
-    for hydrated_target_with_origin in targets_with_origins:
-        target = hydrated_target_with_origin.target
-        if _is_exported(target):
-            exported_targets.append(ExportedTarget(target))
-        elif isinstance(hydrated_target_with_origin.origin, SingleAddress):
-            explicit_nonexported_targets.append(target)
+    explicit_nonexported_targets: List[Target] = []
+
+    for target_with_origin in targets_with_origins:
+        tgt = target_with_origin.target
+        if _is_exported(tgt):
+            exported_targets.append(ExportedTarget(tgt))
+        elif isinstance(target_with_origin.origin, SingleAddress):
+            explicit_nonexported_targets.append(tgt)
     if explicit_nonexported_targets:
         raise TargetNotExported(
             "Cannot run setup.py on these targets, because they have no `provides=` clause: "
-            f'{", ".join(so.adaptor.address.reference() for so in explicit_nonexported_targets)}'
+            f'{", ".join(so.address.reference() for so in explicit_nonexported_targets)}'
         )
 
     if options.values.transitive:
         # Expand out to all owners of the entire dep closure.
-        tht = await Get[TransitiveHydratedTargets](
-            Addresses(et.hydrated_target.adaptor.address for et in exported_targets)
+        transitive_targets = await Get[TransitiveTargets](
+            Addresses(et.target.address for et in exported_targets)
         )
         owners = await MultiGet(
-            Get[ExportedTarget](OwnedDependency(ht)) for ht in tht.closure if is_ownable_target(ht)
+            Get[ExportedTarget](OwnedDependency(tgt))
+            for tgt in transitive_targets.closure
+            if is_ownable_target(tgt)
         )
-        exported_targets = list(set(owners))
+        exported_targets = list(FrozenOrderedSet(owners))
```

Comment:
This is so that we preserver order.

Reply:
How did we not need to preserve order before? 

Is this just to get cache hits, or did correctness depend on the order?

Reply:
I only added it for more cache hits. Don't think we actually need to preserver order afaict.